### PR TITLE
(feature): Add allowExtraFields configuration to TypeScript generators

### PR DESCRIPTION
**Pull request by amckinney**
This adds the `allowExtraFields` configuration option to the TypeScript generators. This permits the user to include extra fields in the request and/or response wire representation rather than being stripped.

When enabled, this makes _every_ request/response allow extra fields. We will later follow-up with a more granular option that can be specified on individual types in the OpenAPI / Fern definitions directly.

### CODE REVIEW

File: generators/typescript/express/express-endpoint-type-schemas-generator/src/GeneratedExpressEndpointTypeSchemasImpl.ts [hunk 1]
```diff
@@ -173,6 +195,17 @@ export class GeneratedExpressEndpointTypeSchemasImpl implements GeneratedExpress
...
                     throw new Error("No response schema was generated");
                 }
+                if (this.allowExtraFields) {
+                    return this.generatedResponseSchema
+                        .getReferenceToZurgSchema(context)
+                        .jsonOrThrow(referenceToParsedResponse, {
+                            unrecognizedObjectKeys: "passthrough",
+                            allowUnrecognizedEnumValues: true,
+                            allowUnrecognizedUnionMembers: true,
+                            skipValidation: true,
```
Comment (dsinghvi) [inline generators/typescript/express/express-endpoint-type-schemas-generator/src/GeneratedExpressEndpointTypeSchemasImpl.ts:205]: should we be skipping validation here?

File: generators/typescript/express/express-error-schema-generator/src/GeneratedExpressErrorSchemaImpl.ts [hunk 2]
```diff
@@ -59,23 +68,43 @@ export class GeneratedExpressErrorSchemaImpl
...
         switch (this.type.type) {
             case "named":
+                if (this.allowExtraFields) {
+                    return context.typeSchema
+                        .getSchemaOfNamedType(this.type, { isGeneratingSchema: false })
+                        .jsonOrThrow(referenceToBody, {
+                            unrecognizedObjectKeys: "passthrough",
+                            allowUnrecognizedEnumValues: true,
+                            allowUnrecognizedUnionMembers: true,
+                            skipValidation: true,
```
Comment (dsinghvi) [inline generators/typescript/express/express-error-schema-generator/src/GeneratedExpressErrorSchemaImpl.ts:78]: same

File: generators/typescript/express/express-endpoint-type-schemas-generator/src/GeneratedExpressEndpointTypeSchemasImpl.ts (hunk 1 above)
Comment (amckinney) [reply generators/typescript/express/express-endpoint-type-schemas-generator/src/GeneratedExpressEndpointTypeSchemasImpl.ts]: No, good catch - updated with a `getSchemaOptions` function (as discussed).

File: generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { SchemaOptions } from "@fern-typescript/zurg";
+
+export function getSchemaOptions({ allowExtraFields }: { allowExtraFields: boolean }): Required<SchemaOptions> {
```
Comment (amckinney) [inline generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts:3]: @dsinghvi I noticed that we actually set `skipValidation` to `false` and `breadcrumbsPrefix` to `[]` in _every_ instance of this function, so I included those properties here too (simply by specifying `Required<SchemaOptions>`. No `Pick` was necessary after all, but we can easily adapt this later to restrict specifics keys later if need be.

For what it's worth, users can still easier override the spread with any overrides they need like we normally do in migrations, i.e.

```typescript
return this.generatedRequestSchema
    .getReferenceToZurgSchema(context)
    .jsonOrThrow(referenceToParsedRequest, {
        ...getSchemaOptions({
            allowExtraFields: this.allowExtraFields
        }),
        skipValidation: true,
    });
```

File: generators/typescript/express/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.11.0-rc0] - 2024-04-12
+
+- Feature: Add `allowExtraFields` option to permit extra fields in the returned response.
```
Comment (dsinghvi) [inline generators/typescript/express/CHANGELOG.md:10]: can we add an example of consuming this in `generators.yml`?

File: generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts (hunk 3 above)
Comment (dsinghvi) [reply generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts:3]: interesting do you know where we set `skipValidation` to true?

**Review by dsinghvi — APPROVED** [review]

File: generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts (hunk 3 above)
Comment (dsinghvi) [reply generators/typescript/utils/commons/src/codegen-utils/getSchemaOptions.ts:3]: i guess we only do it on response fields which this doesn't change

**Pull request merged**